### ISSUE TITLE
ipython mode captures the key presses

### ISSUE THREAD

**Issue by rplevka** [sloria/doitlive#20]
Not sure, whether this is just me, but whenever I try to use an `ipython` mode,
the interpreter starts to capture my random key presses, instead of the scripted commands.

python-ipython.noarch                3.2.1-7.fc24

```bash
 cat doitlive-demo.sh 
# in session.sh
```ipython
def foo():
	print('bar')

foo()
```

```
$ doitlive play doitlive-demo.sh 
We'll do it live!
STARTING SESSION: Press Ctrl-C at any time to exit.
Press any key to continue ...

w1pko@localhost.localdomain:playground $ ipython
Python 2.7.13 (default, May 10 2017, 20:04:36) 
Type "copyright", "credits" or "license" for more information.

IPython 5.3.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: sadasdasdasdasdasfasfa
------------------------------------------------------------------------
NameError                              Traceback (most recent call last)
<ipython-input-1-0048448acb26> in <module>()
----> 1 sadasdasdasdasdasfasfa

NameError: name 'sadasdasdasdasdasfasfa' is not defined

In [2]: quit
w1pko@localhost.localdomain:playground $ 
FINISHED SESSION

```

The basic `python` mode works well though.

**Comment by sloria** [comment]
Can you verify that doitlive is installed in the same virtualenv as IPython? `which doitlive` and `which ipython` should return paths that are in the same directory.

**Comment by rplevka** [comment]
@sloria, thanks for the reply. I didn't use the virtualenv so they were in the same env.
I better tried to use virtualenv and also faced the same issue:
```bash
(doitlive) [ rplevka@localhost:/tmp ]$ which doitlive
~/.virtualenvs/doitlive/bin/doitlive
(doitlive) [ rplevka@localhost:/tmp ]$ which python
~/.virtualenvs/doitlive/bin/python
(doitlive) [ rplevka@localhost:/tmp ]$ which ipython
~/.virtualenvs/doitlive/bin/ipython
```

**Comment by axocomm** [comment]
In case it helps at all, I'm seeing this as well with IPython 5.0.0 and up. 4.2.1 (and several 4.x versions at least; haven't gone too far back) seems to be alright.

**Comment by sloria** [comment]
@axocomm is correct; only IPython<5.0 is supported at the moment. Newer versions of IPython rely on [prompt_toolkit](https://github.com/jonathanslenders/python-prompt-toolkit), which complicates integration with doitlive. 

I would welcome any help getting the code up-to-date with newer versions of IPython!

**Comment by sloria** [comment]
I've got a decent start on this. Will have a WIP PR up soon.

**Comment by sloria** [comment]
This is now fixed in 3.0.0
